src/SFA.DAS.LevyTransferMatching.Api/Controllers/PledgesController.cs

```
@@ -114,6 +117,26 @@ public async Task<IActionResult> CreatePledge(long accountId, [FromBody]CreatePl
             return result;
         }
 
+        [HttpPost]
+        [Route("pledges/close/{pledgeId}")]
+        [ProducesResponseType((int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> ClosePledge(int pledgeId)
+        {
+            var commandResult = await _mediator.Send(new ClosePledgeCommand
```

The other controllers are pretty inconsistent, but I wonder if we could just throw an exception from within the application layer and allow it to bubble up, rather than having a return type tell us whether it succeeded or not?

---

This is now updated:  The application layer is now throwing a custom exception and the controller method is catching it and returning the NoFound response.